### PR TITLE
Task 585 Preposition

### PR DESCRIPTION
**Pull request by cosmicishan**
Task 585_preposition_text_generation -> Given two words, you have to generate which preposition connects both. 

Please review.

### CODE REVIEW

**Comment by swarooprm** [comment]
@kushalchawla, wondering if you can help review this PR?

**Comment by cosmicishan** [comment]
Also, please suggest to me another task I can create from this dataset.
Thanks in advance!!

**Comment by danyaljj** [comment]
I am trying to better understand the task. 
For this instance, 
```
        {
            "input": "Word1: trader \n Word2: stock",
            "output": [
                "on"
            ]
        },
```
why is it "on" and not "of"? (i.e., trader of stock).

**Comment by cosmicishan** [comment]
> I am trying to better understand the task.
> For this instance,
> 
> ```
>         {
>             "input": "Word1: trader \n Word2: stock",
>             "output": [
>                 "on"
>             ]
>         },
> ```
> 
> why is it "on" and not "of"? (i.e., trader of stock).

That's how it's present in the dataset. What do you suggest I do here?

**Comment by danyaljj** [comment]
Is there a way to filter out high-quality instances of the task?

**Comment by cosmicishan** [comment]
> Is there a way to filter out high-quality instances of the task?

I fixed that. Please review again.

**Comment by danyaljj** [comment]
> > Is there a way to filter out high-quality instances of the task?
> 
> I fixed that. Please review again.

Thanks! Could you briefly describe what kind of filtering did you apply?

**Comment by cosmicishan** [comment]
> > > Is there a way to filter out high-quality instances of the task?
> > 
> > 
> > I fixed that. Please review again.
> 
> Thanks! Could you briefly describe what kind of filtering did you apply?

There was multiple files and I only used the one which had better labels.

**Comment by danyaljj** [comment]
What do you mean by "better labels"?

**Comment by cosmicishan** [comment]
> What do you mean by "better labels"?

Like those files have more accurate preposition and other files have multiple words repeating not all prepositions were accurate.

**Comment by danyaljj** [comment]
> > What do you mean by "better labels"?
> 
> Like those files have more accurate preposition and other files have multiple words repeating not all prepositions were accurate.

I understand it this far! :) But how do you measure/estimate "were accurate"? Did you use any quality/confidence scores in the dataset files or just spot-checking?

**Comment by cosmicishan** [comment]
> > > What do you mean by "better labels"?
> > 
> > 
> > Like those files have more accurate preposition and other files have multiple words repeating not all prepositions were accurate.
> 
> I understand it this far! :) But how do you measure/estimate "were accurate"? Did you use any quality/confidence scores in the dataset files or just spot-checking?

No, I just use my own intuition and checked it manually. There were 4 files and I choose the one which was containing more accurate prepositions.

File: tasks/task585_preposition_text_generation.json [hunk 1]
```diff
@@ -0,0 +1,11472 @@
...
+    "Contributors": [
+        "Ishan Purohit"
+    ],
+    "Source": [
+        "Prepositional Paraprasing (Dataset - https://www.cse.iitb.ac.in/~girishp/nc-dataset/) (Paper - https://aclanthology.org/2020.findings-emnlp.386.pdf)"
+    ],
+    "Categories": [
+        "Text Generation"
+    ],
+    "Definition": "In this task, you're given inputs of two words and your task is to predict the correct preposition that connects these two words. This involves rewriting the noun compound as a paraphrase that conveys its meaning explicitly, e.g., orange juice: \u201cjuice made from the orange\u201d or \u201cjuice with orange flavor\u201d.",
```
Comment (danyaljj) [inline tasks/task585_preposition_text_generation.json]: Drop the Unicode characters in the definition. (e.g., `\u201c`). Also, list the possible outputs (`of`, etc).

Comment (cosmicishan) [reply tasks/task585_preposition_text_generation.json]: > Drop the Unicode characters in the definition. (e.g., `\u201c`). Also, list the possible outputs (`of`, etc).

But in the dataset, only one output is given.

Comment (danyaljj) [reply tasks/task585_preposition_text_generation.json]: Maybe I didn't say it right; in the description state what are the possible outputs ("of", "with", etc). Does that make sense?

Comment (cosmicishan) [reply tasks/task585_preposition_text_generation.json]: > Maybe I didn't say it right; in the description state what are the possible outputs ("of", "with", etc). Does that make sense?

Sorry for the misunderstanding. I have fixed that now and corrected the definition.

**Pull request merged**